### ISSUE TITLE
networkd: VRF: Assertion 'h' failed at ../systemd-stable/src/basic/hashmap.c:1265, function hashmap_put()

### ISSUE THREAD

**Issue by k0ste** [systemd/systemd#9150]
**systemd version the issue has been seen with**
> 238.76

```
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN +PCRE2 default-hierarchy=hybrid
```

**Used distribution**
 > Arch Linux (Linux linux01 4.16.12-1-ARCH #1 SMP PREEMPT Fri May 25 23:30:31 UTC 2018 x86_64 GNU/Linux)

**Expected behaviour you didn't see**
 > VRF interfaces is created by `networkd` and in up state.

**Unexpected behaviour you saw**
 > `networkd` failed to start: `Assertion 'h' failed at ../systemd-stable/src/basic/hashmap.c:1265, function hashmap_put(). Aborting.`

### networkd configuration:

dummy0.netdev:
```
[NetDev]
Name=dummy0
Kind=dummy
```
dummy0.network:
```
[Match]
Name=dummy0

[Network]
Address=100.100.100.1/24
VRF=vrf100
```
vrf100.netdev:
```
[NetDev]
Name=vrf100
Kind=vrf

[VRF]
Table=100
```
vrf100.network:
```
[Match]
Name=vrf100
```
journal
```
Jun 01 08:00:13 linux01 systemd[1]: Starting Network Service...
Jun 01 08:00:13 linux01 systemd-networkd[384]: Bus n/a: changing state UNSET → OPENING
Jun 01 08:00:13 linux01 systemd-networkd[384]: Bus n/a: changing state OPENING → AUTHENTICATING
Jun 01 08:00:13 linux01 systemd-networkd[384]: Failed to open configuration file '/etc/systemd/networkd.conf': No such file or directory
Jun 01 08:00:13 linux01 systemd-networkd[384]: timestamp of '/etc/systemd/network' changed
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /etc/systemd/network/dummy0.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /etc/systemd/network/vrf100.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /usr/lib/systemd/network/99-default.link, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /usr/lib/systemd/network/80-container-vz.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /usr/lib/systemd/network/80-container-ve.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Ignoring /usr/lib/systemd/network/80-container-host0.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[384]: Assertion 'h' failed at ../systemd-stable/src/basic/hashmap.c:1265, function hashmap_put(). Aborting.
Jun 01 08:00:13 linux01 systemd[1]: Created slice system-systemd\x2dcoredump.slice.
Jun 01 08:00:13 linux01 systemd[1]: Started Process Core Dump (PID 385/UID 0).
Jun 01 08:00:13 linux01 systemd[1]: systemd-networkd.service: Main process exited, code=dumped, status=6/ABRT
Jun 01 08:00:13 linux01 systemd[1]: systemd-networkd.service: Failed with result 'core-dump'.
Jun 01 08:00:13 linux01 systemd[1]: Failed to start Network Service.
Jun 01 08:00:13 linux01 systemd[1]: systemd-networkd.service: Service has no hold-off time, scheduling restart.
Jun 01 08:00:13 linux01 systemd[1]: systemd-networkd.service: Scheduled restart job, restart counter is at 1.
Jun 01 08:00:13 linux01 sudo[381]: pam_unix(sudo:session): session closed for user root
Jun 01 08:00:13 linux01 systemd[1]: Stopped Network Service.
Jun 01 08:00:13 linux01 systemd[1]: Starting Network Service...
Jun 01 08:00:13 linux01 systemd-networkd[387]: Bus n/a: changing state UNSET → OPENING
Jun 01 08:00:13 linux01 systemd-networkd[387]: Bus n/a: changing state OPENING → AUTHENTICATING
Jun 01 08:00:13 linux01 systemd-networkd[387]: Failed to open configuration file '/etc/systemd/networkd.conf': No such file or directory
Jun 01 08:00:13 linux01 systemd-networkd[387]: timestamp of '/etc/systemd/network' changed
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /etc/systemd/network/dummy0.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /etc/systemd/network/vrf100.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /usr/lib/systemd/network/99-default.link, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /usr/lib/systemd/network/80-container-vz.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /usr/lib/systemd/network/80-container-ve.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Ignoring /usr/lib/systemd/network/80-container-host0.network, because it's not a regular file with suffix .netdev.
Jun 01 08:00:13 linux01 systemd-networkd[387]: Assertion 'h' failed at ../systemd-stable/src/basic/hashmap.c:1265, function hashmap_put(). Aborting.
Jun 01 08:00:13 linux01 systemd-coredump[386]: Process 384 (systemd-network) of user 193 dumped core.
                                               
                                               Stack trace of thread 384:
                                               #0  0x00007f4aa97a986b raise (libc.so.6)
                                               #1  0x00007f4aa979440e abort (libc.so.6)
                                               #2  0x00007f4aa93eae5c log_assert_failed_realm (libsystemd-shared-238.so)
                                               #3  0x00007f4aa943a632 hashmap_put (libsystemd-shared-238.so)
                                               #4  0x0000563cad5e6107 n/a (systemd-networkd)
                                               #5  0x0000563cad5ef799 n/a (systemd-networkd)
                                               #6  0x00007f4aa93c72dd config_parse (libsystemd-shared-238.so)
                                               #7  0x0000563cad5f2376 n/a (systemd-networkd)
                                               #8  0x0000563cad5dbfb2 n/a (systemd-networkd)
                                               #9  0x00007f4aa979606b __libc_start_main (libc.so.6)
                                               #10 0x0000563cad5dd25a n/a (systemd-networkd)
Jun 01 08:00:13 linux01 systemd[1]: Started Process Core Dump (PID 388/UID 0).
Jun 01 08:00:14 linux01 systemd[1]: systemd-networkd.service: Main process exited, code=dumped, status=6/ABRT
Jun 01 08:00:14 linux01 systemd[1]: systemd-networkd.service: Failed with result 'core-dump'.
Jun 01 08:00:14 linux01 systemd[1]: Failed to start Network Service.
```

**Comment by yuwata** [comment]
@andir #6704 seems broken. `config_parse_route_table()` takes `Network` object instead of `uint32_t`.

**Comment by yuwata** [comment]
Fix is waiting in #9152.
